packages/nx/src/tasks-runner/task-env.ts

```diff
@@ -124,7 +124,10 @@ function getNxEnvVariablesForTask(
 
 function loadDotEnvFilesForTask(
   task: Task,
-  environmentVariables: NodeJS.ProcessEnv
+  environmentVariables: NodeJS.ProcessEnv,
+  envExpand = process.env.NX_ENV_EXPAND !== undefined
```

Comment:
this doesn't need to be a argument to this function. You can remove it and use the logic suggested in the if branch